src/IO/H5/VolumeData.cpp

```diff
@@ -159,6 +165,335 @@ void append_element_extents_and_connectivity(
     }
   }
 }
+// Given a std::vector of grid_names, computes the number of blocks that exist
+// and also returns a std::vector of block numbers that is a one-to-one mapping
+// to each element in grid_names. The returned tuple is of the form
+// [number_of_blocks, block_number_for_each_element, sorted_element_indices].
+// number_of_blocks is equal to the number of blocks in the domain.
+// block_number_for_each_element is a std::vector with length equal to the total
+// number of grid names. sorted_element_indices is a std::vector<std::vector>
+// with length equal to the number of blocks in the domain, since each subvector
+// represents a given block. These subvectors are of a length equal to the
+// number of elements which belong to that corresponding block.
+std::tuple<size_t, std::vector<size_t>, std::vector<std::vector<size_t>>>
+compute_and_organize_block_info(const std::vector<std::string>& grid_names) {
+  std::vector<size_t> block_number_for_each_element;
+  std::vector<std::vector<size_t>> sorted_element_indices;
+  block_number_for_each_element.reserve(grid_names.size());
+
+  // Fills block_number_for_each_element
+  for (const std::string& grid_name : grid_names) {
+    size_t end_position = grid_name.find(',', 1);
+    block_number_for_each_element.push_back(
+        static_cast<size_t>(std::stoi(grid_name.substr(2, end_position))));
+  }
+
+  const auto max_block_number =
+      *std::max_element(block_number_for_each_element.begin(),
+                        block_number_for_each_element.end());
+  auto number_of_blocks = max_block_number + 1;
+  sorted_element_indices.reserve(number_of_blocks);
+
+  // Properly sizes subvectors of sorted_element_indices
+  for (size_t i = 0; i < number_of_blocks; ++i) {
+    std::vector<size_t> sizing_vector;
+    auto number_of_elements_in_block =
+        static_cast<size_t>(std::count(block_number_for_each_element.begin(),
+                                       block_number_for_each_element.end(), i));
+    sizing_vector.reserve(number_of_elements_in_block);
+    sorted_element_indices.push_back(sizing_vector);
+  }
+
+  // Organizing grid_names by block
+  for (size_t i = 0; i < block_number_for_each_element.size(); ++i) {
+    sorted_element_indices[block_number_for_each_element[i]].push_back(i);
+  }
+
+  return std::make_tuple(number_of_blocks,
+                         std::move(block_number_for_each_element),
+                         std::move(sorted_element_indices));
+}
+
+// Sort input by block
```

Comment:
Can you expand this a bit more? I found it a bit surprising that there wasn't a `std::sort` call. Maybe describe how each input relates to the output?